File: main/HappyParserPy/bitmap.py
from PIL import Image

def read_content(x, o):
    return x[o], x[o+1], x[o+2]

def display(content, show: bool):
    offset=0
    img = Image.new('RGB', (content.width, content.height), "magenta")
    pixels = img.load()
    for j in range(img.size[1]):
        for i in range(img.size[0]):
            pixels[i, j] = read_content(content.content, offset)
            offset += 3
    if show:
        img.show()

File: main/CaffWeb/CaffDal/Services/Parser/bitmap.cs
﻿using SkiaSharp;
using SkiaSharp.Views.Desktop;

namespace CaffDal.Services.Parser
{
    public static class Parser
    { 
    public static Tuple<byte, byte, byte> read_content(byte[] x, int o)
    {
        return Tuple.Create(x[o], x[o + 1], x[o + 2]);
    }

    public static byte[] display(Ciff content)
    {
        var offset = 0;
        SKBitmap img = new SKBitmap(width: content.Width, height: content.Height);
            for (int i = 0; i < img.Height; i++)
            {
                for (int j = 0; j< img.Width;j ++)
                {
                var color = read_content(content.RawCiff, offset);
                    img.SetPixel(j, i, new SKColor(red: color.Item1, green: color.Item2, blue: color.Item3));
                offset += 3;
                }
            }
            using (MemoryStream stream = new MemoryStream())
            {
                img.Encode(stream, SKEncodedImageFormat.Jpeg, Int32.MaxValue);
                return stream.ToArray();
            }

        }
  }
}
